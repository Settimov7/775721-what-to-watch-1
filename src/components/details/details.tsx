import * as React from 'react';

export const Details = (props) => {
  const {director, starring, runTime, genre, releasedYear} = props;

  return(
    <div className="movie-card__text movie-card__row">
      <div className="movie-card__text-col">
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">Director</strong>
          <span className="movie-card__details-value">{director}</span>
        </p>
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">Starring</strong>

          {starring.map((actor, index) => {
            const isLastActor = index === starring.length -1;

            if(isLastActor) {
              return <span key={actor} className="movie-card__details-value">{actor}</span>;
            }

            return (
              <span key={actor} className="movie-card__details-value">
                {`${actor},`}<br />
              </span>
            )
          })}
        </p>
      </div>

      <div className="movie-card__text-col">
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">Run Time</strong>
          <span className="movie-card__details-value">{runTime}</span>
        </p>
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">Genre</strong>
          <span className="movie-card__details-value">{genre}</span>
        </p>
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">Released</strong>
          <span className="movie-card__details-value">{releasedYear}</span>
        </p>
      </div>
    </div>
  );
};
