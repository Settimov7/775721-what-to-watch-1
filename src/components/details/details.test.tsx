import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Details} from './details';

it(`Details correctly renders`, () => {
  const props = {
    director: `director`,
    starring: [`person-1`, `person-2`, `person-3`],
    runTime: 99,
    genre: `Genre`,
    releasedYear: 2019,
  };
  const details = renderer.create(
    <Details {...props}/>
  ).toJSON();

  expect(details).toMatchSnapshot();
});
