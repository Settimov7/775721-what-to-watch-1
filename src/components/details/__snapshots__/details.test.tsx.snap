// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details correctly renders 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        director
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        person-1,
        <br />
      </span>
      <span
        className="movie-card__details-value"
      >
        person-2,
        <br />
      </span>
      <span
        className="movie-card__details-value"
      >
        person-3
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        99
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Genre
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2019
      </span>
    </p>
  </div>
</div>
`;
